ci: knex@0.21.8 cannot be tested with node v8

**Describe the bug**

The knex package has dropped support for Node 8 and introduced code that breaks Node 8.  This has broken the build.  No CI can pass until this is resolved.

    /home/travis/build/elastic/apm-agent-nodejs/node_modules/knex/lib/dialects/postgres/schema/columncompiler.js:7
    const commentEscapeRegex = /(?<!')'(?!')/g;
    SyntaxError: Invalid regular expression: /(?<!')'(?!')/: Invalid group

For the record the full error in CI looks like this (e.g. from https://travis-ci.org/github/elastic/apm-agent-nodejs/jobs/739313279):

```
running (cwd: ./): node --require /home/travis/build/elastic/apm-agent-nodejs/test/_promise_rejection.js test/instrumentation/modules/pg/knex.js
TAP version 13
# knex.select().from('test')
/home/travis/build/elastic/apm-agent-nodejs/node_modules/knex/lib/dialects/postgres/schema/columncompiler.js:7
const commentEscapeRegex = /(?<!')'(?!')/g;
                           ^
SyntaxError: Invalid regular expression: /(?<!')'(?!')/: Invalid group
    at Object.<anonymous> (/home/travis/build/elastic/apm-agent-nodejs/node_modules/knex/lib/dialects/postgres/schema/columncompiler.js:7:28)
    at Module._compile (module.js:624:30)
    at Object.Module._extensions..js (module.js:635:10)
    at Module.load (module.js:545:32)
    at tryModuleLoad (module.js:508:12)
    at Function.Module._load (module.js:500:3)
    at Module.require (module.js:568:17)
    at Module.Hook._require.Module.require (/home/travis/build/elastic/apm-agent-nodejs/node_modules/require-in-the-middle/index.js:80:39)
    at require (internal/module.js:11:18)
    at Object.<anonymous> (/home/travis/build/elastic/apm-agent-nodejs/node_modules/knex/lib/dialects/postgres/index.js:9:24)
    at Module._compile (module.js:624:30)
    at Object.Module._extensions..js (module.js:635:10)
    at Module.load (module.js:545:32)
    at tryModuleLoad (module.js:508:12)
    at Function.Module._load (module.js:500:3)
    at Module.require (module.js:568:17)
    at Module.Hook._require.Module.require (/home/travis/build/elastic/apm-agent-nodejs/node_modules/require-in-the-middle/index.js:80:39)
    at require (internal/module.js:11:18)
    at Knex (/home/travis/build/elastic/apm-agent-nodejs/node_modules/knex/lib/knex.js:44:15)
    at Client.<anonymous> (/home/travis/build/elastic/apm-agent-nodejs/test/instrumentation/modules/pg/knex.js:135:12)
    at Object.onceWrapper (events.js:314:30)
    at emitNone (events.js:105:13)
    at Client.emit (events.js:207:7)
    at process.nextTick (/home/travis/build/elastic/apm-agent-nodejs/node_modules/pg/lib/client.js:154:14)
    at _combinedTickCallback (internal/process/next_tick.js:131:7)
    at process._tickCallback (internal/process/next_tick.js:180:9)
/home/travis/build/elastic/apm-agent-nodejs/test/test.js:129
    if (err) throw err
             ^
Error: non-zero error code
    at ChildProcess.<anonymous> (/home/travis/build/elastic/apm-agent-nodejs/test/test.js:37:19)
    at emitTwo (events.js:125:13)
    at ChildProcess.emit (events.js:213:7)
    at maybeClose (internal/child_process.js:927:16)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:211:5)
npm ERR! Test failed.  See above for more details.
The command "npm test" exited with 1.
```

knex dropped node v8 support in 0.21.0.  knex@0.21.8 (released yesterday) finally broke node v8 syntax.

Alan and I chatted yesterday and have decided for now to add a guard in the knex test file that will effectively result in not testing knex with node v8 (it will still test with node v10 and later versions) -- at least for the "test with the latest version of modules" testing that is done for PRs and commits.   The nightly "tav" (test-all-versions) tests will still test knex@0.20.0 with node v8 (per https://github.com/elastic/apm-agent-nodejs/blob/3a845efe2fc3278135a0ab8ac9a4aa5f832605f4/.tav.yml#L88-L97).

--- a/test/instrumentation/modules/pg/knex.js
+++ b/test/instrumentation/modules/pg/knex.js
@@ -13,6 +13,11 @@ var agent = require('../../../..').start({
 var knexVersion = require('knex/package').version
 var semver = require('semver')
 
+if (semver.gte(knexVersion, '0.21.0') && semver.lt(process.version, '10.0.0')) {
+  // Skip these tests: knex@0.21.x dropped support for node v8.
+  process.exit(0)
+}
+
 var Knex = require('knex')
 var test = require('tape')
 

